### DISCUSSION TITLE
Recommended way of passing Airflow context to KubernetesPodOperator?

### DISCUSSION THREAD

**Discussion by RonaldGalea** [apache/airflow#35110] [Q&A] [answered]
With the Taskflow API for regular python tasks this is nicely achievable via:

```py
@task
def my_fn(**context):
    # context accessible
```

However, with the equivalent decorator for the KubernetesPodOperator task, the context seems not to be there:

```py
@task.kubernetes
def my_fn_to_run_in_pod(**context):
    # nothing, context is {}
```

Am I perhaps doing something wrong, or is this not expected to work in this manner? What would be the correct way to achieve this? There is one StackOverflow [solution](https://stackoverflow.com/questions/65911405/get-context-from-pod-launched-with-airflow-kubernetespodoperator?rq=4) that hooks into `.execute()`, where the context is seemingly accessible, and sets information for the pod via environment variables. Still, I'm not sure it's correct to make use of internal implementation like this.

Many thanks.

**Comment by potiuk** [comment, marked as answer]
AFAIK. There is no better solution and the proposed solution is not really using internals - it uses only public APIS described in https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html (and extending an operator to add functionality you need is absolutely first-class citizen of Airflow).

BTW. If you would like to contribute back and propose a PR to make it more "out-of-the-box" - feel free. Airflow has many contributors and if you can take a leadership to add the PR/tests to make it more "built-in" that would be cool. It would likely require more comprehensive solution - only some, selected properties of the context are passed so probably you could make it more "complete".

Alternatively, if you find it good-enough for your case, maybe a good idea is to contribute a  documentation PR describing it as one of the possible solutions. Once you do so, you will get maintainers to take a look and help to review it, so I think it's a good idea to give back for the free software you use :).

And it's rather easy - just click "suggest a change on this page" in the docs page of K8S provider and PR will be opened for you that you will be able to contribute. It's a great start to contribute such thing.
